tests/components/lg_soundbar/test_config_flow.py

```diff
@@ -209,6 +208,208 @@ def temescal_side_effect(addr, port, callback):
     mock_uuid_q.get.assert_called_once()
 
 
+async def test_form_uuid_missing_from_mac_info(hass):
+    """Test we get the form, but uuid is missing from the initial get_mac_info function call."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    with patch(
+        "homeassistant.components.lg_soundbar.config_flow.temescal"
+    ) as mock_temescal, patch(
+        "homeassistant.components.lg_soundbar.async_setup_entry", return_value=True
+    ) as mock_setup_entry:
+        tmock = mock_temescal.temescal
+        tmock.return_value = Mock()
+        instance = tmock.return_value
+
+        def temescal_side_effect(addr, port, callback):
+            mac_info = {"msg": "MAC_INFO_DEV"}
+            instance.get_mac_info.side_effect = lambda: callback(mac_info)
+            product_info = {"msg": "PRODUCT_INFO", "data": {"s_uuid": "uuid"}}
+            instance.get_product_info.side_effect = lambda: callback(product_info)
+            info = {"msg": "SPK_LIST_VIEW_INFO", "data": {"s_user_name": "name"}}
+            instance.get_info.side_effect = lambda: callback(info)
+            return DEFAULT
+
+        tmock.side_effect = temescal_side_effect
+
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_HOST: "1.1.1.1",
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == "create_entry"
+    assert result2["title"] == "name"
+    assert result2["data"] == {
+        CONF_HOST: "1.1.1.1",
+        CONF_PORT: DEFAULT_PORT,
+    }
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_uuid_not_provided_by_api(hass):
+    """Test we get the form, but uuid is missing from the all API messages."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    mock_uuid_q = MagicMock()
+    mock_name_q = MagicMock()
+
+    with patch(
+        "homeassistant.components.lg_soundbar.config_flow.temescal"
+    ) as mock_temescal, patch(
+        "homeassistant.components.lg_soundbar.config_flow.Queue"
+    ) as mock_q, patch(
+        "homeassistant.components.lg_soundbar.config_flow.LGSoundbarConfigFlow.async_set_unique_id"
+    ) as mock_set_unique_id, patch(
+        "homeassistant.components.lg_soundbar.async_setup_entry", return_value=True
+    ) as mock_setup_entry:
+        mock_q.side_effect = [mock_uuid_q, mock_name_q]
+        mock_uuid_q.empty.return_value = True
+        mock_uuid_q.get.side_effect = Empty
```

Comment:
Instead of patching the queue, I'd patch the timeout of the queue to 0. This is easy if we define the timeout in a module level constant in the config flow. If needed we can sleep a bit in the callback to make sure the timeout always is faster than the callback.